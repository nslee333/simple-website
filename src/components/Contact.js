export default function Contact() {
  const renderForm = () => {
    return (
      <>
        <form name="contact" data-netlify="true" method="post">
          <input type="hidden" name="form-name" value="contact" />
          <label>
            <div hidden>name</div>
            <div>
              <input
                className="my-2 
                          block 
                          w-full 
                          rounded-md 
                          p-2"
                type="text"
                name="name"
                placeholder="Full Name"
                required={true}
              />
            </div>
          </label>
          <label>
            <div hidden>email</div>
            <div className="my-3">
              <input
                className="my-2 
                          block 
                          w-full 
                          rounded-md 
                          p-2"
                type="email"
                name="email"
                placeholder="Email"
                required={true}
              />
            </div>
          </label>
          <label>
            <div
              className="my-4
                        block
                        w-full"
            >
              <textarea
                className="my-2 
                            h-[10rem] 
                            w-full 
                            resize-none 
                            rounded-md 
                            p-2"
                type="text"
                name="message"
                placeholder="..."
                required={true}
              />
            </div>
          </label>
          <button
            className="shadow-grey 
                        mb-2 
                        ml-2 
                        mt-4 
                        block
                        h-[2rem] 
                        w-[7rem] 
                        rounded-md 
                        bg-white 
                        text-xl 
                        shadow
                        hover:translate-y-0.5
                        hover:origin-bottom 
                        hover:shadow-none"
            type="submit"
          >
            Send
          </button>
        </form>
      </>
    );
  };

  return (
    <>
      <div className="flex justify-center p-2.5">
        <div
          className="mx-2 
                      mt-5 
                      w-[32rem] 
                      min-w-[18.85rem] 
                      rounded-md
                      bg-white "
          id="contact"
        >
          <h1
            className="m-5 
                        flex 
                        justify-center 
                        text-2xl"
          >
            Contact Us...
          </h1>
          <div
            className="m-5 
                        rounded-md
                        bg-neutral-300 
                        p-5"
          >
            {renderForm()}
          </div>
        </div>
      </div>
    </>
  );
}
