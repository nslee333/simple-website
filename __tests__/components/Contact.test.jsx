import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Contact from "../../src/components/Contact";
import renderer from "react-test-renderer";

test("Contact renders without crashing", () => {
  render(<Contact />);
});

test("Contact renders correctly according to snapshot.", () => {
  const tree = renderer.create(<Contact />).toJSON();

  expect(tree).toMatchSnapshot();
});

test("Name input is present", () => {
  const { getByLabelText } = render(<Contact />);

  const nameInput = getByLabelText("name");

  expect(nameInput).toBeInTheDocument();
});

test("Email input is present", () => {
  const { getByLabelText } = render(<Contact />);

  const emailInput = getByLabelText("email");

  expect(emailInput).toBeInTheDocument();
});

test("Check if Sign up button is present", () => {
  const { getByText } = render(<Contact />);

  const signUpButton = getByText("Send");

  expect(signUpButton).toBeInTheDocument();
});
