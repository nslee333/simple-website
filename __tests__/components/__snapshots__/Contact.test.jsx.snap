// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders correctly according to snapshot. 1`] = `
<div
  className="flex justify-center p-2.5"
>
  <div
    className="mx-2  mt-5  w-[32rem]  min-w-[18.85rem]  rounded-md bg-white "
    id="contact"
  >
    <h1
      className="m-5  flex  justify-center  text-2xl"
    >
      Contact Us...
    </h1>
    <div
      className="m-5  rounded-md bg-neutral-300  p-5"
    >
      <div
        data-netlify="true"
        method="post"
        name="contact"
      >
        <input
          name="form-name"
          type="hidden"
          value="contact"
        />
        <label>
          <div
            hidden={true}
          >
            name
          </div>
          <div>
            <input
              className="my-2  block  w-full  rounded-md  p-2"
              name="name"
              placeholder="Full Name"
              required={true}
              type="text"
            />
          </div>
        </label>
        <label>
          <div
            hidden={true}
          >
            email
          </div>
          <div
            className="my-3"
          >
            <input
              className="my-2  block  w-full  rounded-md  p-2"
              name="email"
              placeholder="Email"
              required={true}
              type="email"
            />
          </div>
        </label>
        <label>
          <div
            className="my-4 block w-full"
          >
            <textarea
              className="my-2  h-[10rem]  w-full  resize-none  rounded-md  p-2"
              name="message"
              placeholder="..."
              required={true}
              type="text"
            />
          </div>
        </label>
        <button
          className="shadow-grey  mb-2  ml-2  mt-4  block h-[2rem]  w-[7rem]  rounded-md  bg-white  text-xl  shadow hover:translate-y-0.5 hover:origin-bottom  hover:shadow-none"
          type="submit"
        >
          Send
        </button>
      </div>
    </div>
  </div>
</div>
`;
